app/components/chat/ChatRoom.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import { connect } from 'react-redux'
+import { Link } from 'react-router'
+import Helmet from 'react-helmet'
+import Notification from './Notification'
+import Messages from './Messages'
+import Messenger from './Messenger'
+
+// material ui
+import { Paper } from 'material-ui'
+
+// ChatRoom is a local state component that handles messages to/from
+// patient and therapist
+
+// -=-=-=-=-=-= COMPONENT =-=-=-=-=-=-
+
+export class ChatRoom extends Component {
+
+  constructor(props) {
+    super(props)
+    // local state of user, messages array, and 'wipe clean' message
+    this.state = {
+      user: this.props.user.first_name + ' ' + this.props.user.last_name,
+      messages: [],
+      message: '',
+      notifications: '\xa0'
+    }
+    // pick a unique room for the therapist-patient chat based on patient id
+    this.room = this.props.user.role === 'patient' ?
+      this.props.user.id.toString() : this.props.currentPatient.id.toString()
+    // bind our methods to use in render()
+    this.onMessageReceived = this.onMessageReceived.bind(this)
+    this.onMessageSent = this.onMessageSent.bind(this)
+    this.onMessageChange = this.onMessageChange.bind(this)
+    this.onNotification = this.onNotification.bind(this)
+  }
+
+  componentDidMount() {
+    // initialize client socket
+    this.socket = io.connect()
+    // let server know the room to 'speak' in and that a user has entered that room
+    this.socket.emit('userEnter', { room: this.room, user: this.state.user })
+    // workhorse function on client side for message handling
+    this.socket.on('newMessage', this.onMessageReceived)
+    this.socket.on('notification', this.onNotification)
+  }
+
+  componentWillUnmount() {
+    // notify departure on chat close
+    this.socket.emit('userLeave', { room: this.room, user: this.state.user })
+  }
+
+  onMessageReceived(data) {
+    let message = {}
+    if (data.user === this.state.user) {
```

Comment:
Fair point, but we're not sharing user.id's in the message. It's ok though because this.state.user defines data.user, so they should always be the same on the sender's side.